pkg/crd/testdata/cronjob_types.go

```diff
@@ -252,6 +252,24 @@ type CronJobSpec struct {
 
 	// Checks that arrays work when the type contains a composite literal
 	ArrayUsingCompositeLiteral [len(struct{ X [3]int }{}.X)]string `json:"arrayUsingCompositeLiteral,omitempty"`
+
+	// This tests string slice item validation.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=255
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?([.][a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
+	Hosts []string `json:"hosts,omitempty"`
```

Comment:
I think we have to find a way to differentiate if the validation is for the list as a whole or for the individual items in the list. This change just changes things to assume that its for the items. I don't think that is correct and its a breaking change

Reply:
> I don't think that is correct and its a breaking change

I don't think its a breaking change as previously these markers were rejected for slice types.
I also do not see a solid reason to introduce a new set of "items" markers as they will be mutually exclusive with those I change in this PR.